arduino-ide-extension/src/common/protocol/monitor-service.ts

```diff
@@ -39,6 +39,7 @@ export interface MonitorManagerProxyClient {
   getCurrentSettings(board: Board, port: Port): Promise<MonitorSettings>;
   send(message: string): void;
   changeSettings(settings: MonitorSettings): void;
+  setUploadInProgress(value: boolean): void;
```

Comment:
If this call goes through the wire and is not a notification, it should be a promise.

Reply:
it should all be happening frontend

Reply:
the responsibility of "MonitorManagerProxyClientImpl" is a bit muddled, is invokes methods on the backend on behalf of the client, but also the backend invokes its' methods to execute frontend logic, we can improve this